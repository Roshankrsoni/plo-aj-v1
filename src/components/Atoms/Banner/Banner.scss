.banner {
  margin: 2rem;
  position: fixed;
  top: 0;
  left: 50%;
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 40%;

  transform: translateX(-50%);
  z-index: 1001;
}
.banner-text {
  width: calc(100% - 2rem);
  text-align: center;
}
.cross {
  margin-left: 1rem;
  position: relative;
  height: 1rem;
  width: 1rem;
  cursor: pointer;
}

.cross::after {
  content: "";
  position: absolute;
  width: 2px;
  height: 100%;
  background-color: white;

  transform: translateX(0.5rem) rotate(45deg);
}

.cross::before {
  content: "";
  position: absolute;
  width: 2px;
  height: 100%;
  background-color: white;

  transform: translateX(0.5rem) rotate(-45deg);
}

.success-banner {
  padding: 0.5rem 1rem;
  color: white !important;
  background: var(--green);
  border-radius: 0.25rem;
}
.error-banner {
  padding: 0.5rem 1rem;
  color: white !important;
  background: var(--red);
  border-radius: 0.25rem;
}
.neutral-banner {
  padding: 0.5rem 1rem;
  color: white !important;
  background: var(--grey3);
  border-radius: 0.25rem;
}

/* Medium devices (tablets, 768px and up)*/
@media (max-width: 1200px) {
  .banner {
    top: 0;
    left: 0;
    padding: 1rem !important;
    margin-top: 1rem;
    width: calc(100% - 2rem);
    font-weight: bold;

    transform: translateX(-1rem);
  }
}
