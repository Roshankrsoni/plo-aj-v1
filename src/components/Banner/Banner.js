import React from "react";
import "./Banner.css";
const Banner = (props) => {
  return (
    <div
      className={`default-text banner ${props.type}-banner mb-2`}
      data-testid="banner"
    >
      <div className="banner-text" data-testid="banner-text">
        {props.text}
      </div>
      <div
        className="cross"
        onClick={props.closeBanner}
        data-testid="bannerclosebutton"
      ></div>
    </div>
  );
};

export default Banner;
export { Banner };
