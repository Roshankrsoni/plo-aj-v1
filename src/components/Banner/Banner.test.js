import React from 'react';
import { shallow } from 'enzyme';
import Banner from './Banner';

describe('MyComponent', () => {
    it('should render correctly in "debug" mode', () => {
        const component = shallow(<Banner />);
        expect(component).toMatchSnapshot();
    });
});