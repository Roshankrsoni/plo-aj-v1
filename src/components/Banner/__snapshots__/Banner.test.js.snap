// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render correctly in "debug" mode 1`] = `
<div
  className="default-text banner undefined-banner mb-2"
>
  <div
    className="banner-text"
  />
  <div
    className="cross"
  />
</div>
`;
